sdk/trace/attributesmap.go

```diff
@@ -42,7 +45,22 @@ func newAttributesMap(capacity int) *attributesMap {
 	return lm
 }
 
-func (am *attributesMap) add(kv attribute.KeyValue) {
+func (am *attributesMap) Len() int {
+	am.mu.RLock()
+	defer am.mu.RUnlock()
+	return am.evictList.Len()
+}
+
+func (am *attributesMap) DroppedCount() int {
+	am.mu.RLock()
+	defer am.mu.RUnlock()
+	return am.droppedCount
+}
+
+func (am *attributesMap) Add(kv attribute.KeyValue) {
```

Comment:
What is the reason for the visibility changes here and in `evictedqueue.go`?

Reply:
Since the struct is private, it's still technically private. The reason here is to differentiate between methods that can be called by other structs within the same package, and methods that should only be called within the struct itself.

Reply:
Unexported structs can still be made available outside of their package and exported methods on them can be invoked from outside the package.  Within a package there are no restrictions on what types or methods may be used, so it is not necessary to export a method to make it available to methods on other types within the package.  This is part of why I am usually very careful to use the terms (un)exported and not private/public, since they have very different denotations.